src/reify.ml4

```diff
@@ -1582,6 +1603,21 @@ struct
         (*                                   typ; t'|])) *)
         k (evm, Term.mkApp (texistT_typed_term, [|typ; t'|]))
       | _ -> monad_failure "tmUnquote" 1
+    else if Term.eq_constr coConstr tmUnquoteTyped then
+      match args with
+      | typ::t::[] ->
+        let (evm, t) = reduce_all env evm t in
+        let evdref = ref evm in
+        let t' = denote_term evdref t in
+        (* todo: investigate why the following bugs *)
```

Comment:
I commented the type checking in `tmUnquoteTyped` but the bug seems to have disappeared.
Do we prefer `tmUnquoteTyped` or `tmUnquoteUnsafe`?

Reply:
UnquoteUnsafe ? I’d rather say tmUnquoteAtType

Reply:
By Unsafe I mean we remove the type checking of the unquoted term (as it is for the moment)

Reply:
Eventually, we would like to claim that using the monad does not jeopardize Coq's consistency. So, eventually, it seems necessary to do the dynamic type checking in `tmUnquoteAtType`, doesn't it?

Reply:
Yes. I didn’t think of that. Indeed if we ever want to trust templatemonad programs we need that.